### PR TITLE
Tests: Extract e2e test utils to their own package

### PR DESCRIPTION
**Pull request by gziolo**
## Description
Part of #12313.

I did some final tweaks for the code organization related to e2e tests:
- Extracted e2e test utils to their own package with all utils exposed for general usage as `@wordpress/e2e-test-utils`;
- Renamed `@wordpress/tests-e2e` to `@wordpress/e2e-tests` to align with the newly extracted package name.
- Moved all tests that verify the logic that depends on extensibility and test plugins to their own folder: `specs/plugins/`.
- Reintroduced a simplified config file for e2e tests to make it easier to use it outside of Gutenberg.

## How has this been tested?

`npm run test-e2e`
or 
`./node_modules/.bin/wp-scripts test-e2e --config ./packages/e2e-tests/jest.config.js`

### CODE REVIEW

File: packages/e2e-tests/specs/adding-blocks.test.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ import {
 	insertBlock,
 	getEditedPostContent,
 	pressKeyTimes,
-} from '../support/utils';
+} from '@wordpress/e2e-test-utils';
```
Comment (gziolo) [inline packages/e2e-tests/specs/adding-blocks.test.js]: Update code comments to `WordPress dependencies` in all files :)

File: packages/e2e-tests/specs/blocks/__snapshots__/quote.test.js.snap [hunk 2]
```diff
@@ -58,7 +58,11 @@ exports[`Quote can be converted to paragraphs and renders a paragraph for the ci
 <!-- /wp:paragraph -->"
 `;
 
-exports[`Quote can be converted to paragraphs and renders a void paragraph if both the cite and quote are void 1`] = `""`;
+exports[`Quote can be converted to paragraphs and renders a void paragraph if both the cite and quote are void 1`] = `
```
Comment (gziolo) [inline packages/e2e-tests/specs/blocks/__snapshots__/quote.test.js.snap]: Something went wrong here ...

File: packages/e2e-tests/specs/plugins/__snapshots__/meta-attribute-block.test.js.snap [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Block with a meta attribute Should persist the meta attribute properly 1`] = `"<!-- wp:test/test-meta-attribute-block /-->"`;
```
Comment (gziolo) [inline packages/e2e-tests/specs/plugins/__snapshots__/meta-attribute-block.test.js.snap]: It looks like I didn't rename the file properly.

**Comment by gziolo** [comment]
@youknowriad - I haven't found a good way to run those tests from withing `node_modules` subfolder. We probably will have to use symlink with WordPress core to make it work. See related issues:
- https://github.com/facebook/jest/issues/2145
- https://github.com/facebook/create-react-app/issues/1042
- https://github.com/facebook/jest/issues/1995
There is much more. In general Jest by default ignores everything in `node_modules` folder to speed up bootstrap. It makes a lot of sense so I wouldn't fight against it but rather use symlinks or copy tests to a different location.

**Comment by gziolo** [comment]
@gwwar @Copons - not sure if you are still looking into increasing code reuse from core for your e2e tests setup, but this PR tries to expose all test utils as a npm package. We exposed npm scripts which runs the e2e tests setup in one of the previous PRs. Once this code is published to npm, it should be easier to replicate a similar testing workflow.

**Comment by gziolo** [comment]
@ryelle and @greatislander - we are close to having all parts of Gutenberg's setup exposed to npm. It may take a few more days (hopefully not weeks) to have it sorted out. It should drastically simplify the way you can use it outside of Gutenberg. I'm going to work on having tooling for a11y included in this setup as my next task.

**Review by jorgefilipecosta — APPROVED** [review]
Currently, we were not able to execute a single end to end test case (e.g.: executing: `npm run test-e2e packages/e2e-tests/specs/sidebar.test.js`) this PR fixes that. Although I don't have deep knowledge of the changes they make sense to me 👍

**Pull request merged**